docs/Containerfile

```diff
@@ -0,0 +1,17 @@
+FROM registry.fedoraproject.org/fedora:41
+
+RUN dnf install -y \
+    gcc \
+    gcc-c++ \
```

Comment:
Uh, so let me get this straight: rendering our site requires C and C++ compilers? :flushed: :laughing: 
I can see it all breaks if you just remove them, I guess it builds some Ruby extensions from source, but seems pretty wacko on the outset. :smile:

Reply:
Indeed :laughing: There's a lot going on, including C++ compilation, when installing the required gems. Somewhat I feel more comfortable to just do it a container, hence the `JEKYLL_BIN` option that defaults to `podman`.

Of course, people who develop with Ruby likely have this stuff already installed locally or system-wide and thus don't need (or want, really) to duplicate all that data once again in a container, so there's the `bundler` option too. This is also useful for when you already build RPM inside an interactive container (like Toolbox) and want to run "make site" from it.

Reply:
FWIW, some of the required Ruby gems (including Jekyll itself) do come as RPM packages in Fedora, but not all of them. Also, using the "meta gem" github-pages ensures that the correct versions are installed, which is nice too.

Reply:
Yep, a container FTW for sure. There were some pretty hair-raising messages scrolling by in the process of buildign this stuff, such as
> Don't run Bundler as root. Installing your bundle as root will break this
application for all non-root users on this machine.

Reply:
Oh yup, noticed those too :sweat_smile: That should be fine in the container environment of course but is surely hair-raising :smile:

Reply:
Yup. So to confirm, keeping this stuff inside a container unconditionally is a good thing.

Reply:
Nah, maybe not. It's literally 4 or so cmake commands, and can be useful.

Reply:
Oh I managed to gloss over that somehow, looking for something else I guess.
I would really contain this to containers only. It's not a lot of code but the less options there are the less options there are to maintain and worry about, and downloading and installing stuff from the net from our makefiles is not something I'm terribly comfortable with.

Reply:
Fair point. The bundler option isn't downloading anything, though, it's just checking for the gems locally (like we do with any other build deps in cmake). It assumes you have the deps preinstalled, or install them yourself.

That said, indeed, the extra option (+ maintenance) isn't worth the added benefit (of using the native gems if you have them). So I'll drop that. It'll make the whole thing simpler, too.